product_variant_sale_price/models/product_product.py

```diff
@@ -0,0 +1,82 @@
+# Copyright 2016 Sergio Teruel <sergio.teruel@tecnativa.com>
+# License AGPL-3 - See http://www.gnu.org/licenses/agpl-3.0.html
+
+from odoo import api, fields, models
+
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+
+    def _update_fix_price(self, vals):
+        if 'list_price' in vals:
+            self.mapped('product_variant_ids').write({
+                'fix_price': vals['list_price']})
+
+    @api.model
+    def create(self, vals):
+        product_tmpl = super(ProductTemplate, self).create(vals)
+        product_tmpl._update_fix_price(vals)
+        return product_tmpl
+
+    @api.multi
+    def write(self, vals):
+        res = super(ProductTemplate, self).write(vals)
+        for template in self:
+            if not self.env.context.get('skip_update_fix_price', False):
+                template._update_fix_price(vals)
+        return res
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    @api.multi
+    @api.depends('fix_price')
+    def _compute_lst_price(self):
+        uom_model = self.env['uom.uom']
+        for product in self:
+            price = product.fix_price or product.list_price
+            if 'uom' in self.env.context:
+                price = product.uom_id._compute_price(
```

Comment:
you can use else instead of update price

Reply:
I don't understand what you mean. Can you explain it.

Reply:
if 'uom' in self.env.context:
    price = product.uom_id._compute_price( ...
else:
   price = product.fix_price or product.list_price

como estaba antes se asignaba el valor del precio y si se cumple la condición se machaca, con el else, sólo se asigna la variable una vez.

Reply:
Yes but, `_compute_price` method needs that price was set before to compute the new price. Do you agree? 

I think the best option is to leave the code as is.

Reply:
OK, sorry I did not notice that